examples/dataproc-persistent-history-server/README.md

```diff
@@ -0,0 +1,104 @@
+# Dataproc Persistent History Server
+This repo houses the example code for a blog post on using a persistent history
+server to retain infomation about your Spark and Hadoop jobs that ran on short-lived
+clusters.
+
+![Architecture Diagram](img/persistent-history-arch.png)
+
+## Directory structure
+
+- `cluster_templates/`
+  - `history_server.yaml`
+  - `ephemeral_cluster.yaml` 
+- `init_actions`
+  - `disable_history_servers.sh`
+- `workflow_templates`
+  - `spark_mr_workflow_template.yaml`
+- `terraform`
+  - `variables.tf` 
+  - `network.tf` 
+  - `hadoop-subnet.tf` 
+  - `history-server.tf` 
+  - `history-bucket.tf` 
+  - `firewall.tf` 
+
+## Usage
+
+### Terraform
+To spin up the whole example you could simply edit the 
+`terraform.tfvars` file to set the variables to the 
+desired values and run the following commands.
+
+```
+cd terraform
+terraform init
+terraform apply
+```
+
+### Google Cloud SDK
+1.  Replace `PROJECT` with your GCP project id in each file.
+1.  Replace `HISTORY_BUCKET` with your GCS bucket for logs in each file.
+1.  Replace `REGION` with your desired GCP Compute region.
+1.  Manually create the `spark-events` folder in your history bucket.
```

Comment:
Is there a kosher way to create an empty "directory" in a gcs bucket in terraform? Would a 0B object of the same name work? The way spark logging works it needs this directory to already exist.

Reply:
Is there any harm in having a file in the folder? Ex. an empty `.keep` file?

Reply:
I think that will be fine I'll run a test.